querygen/querygen.go

```diff
@@ -87,3 +87,25 @@ func GenClairScanCompletedQuery(index string) string {
 	)
 	return q
 }
+
+// GenReleaseCompletedQuery returns a Splunk query for generating Segment events
+// representing the Release resource success/failure state changes.
+func GenReleaseCompletedQuery(index string) string {
+	q, _ := UserJourneyQueryGen(
+		index,
+		`verb=patch `+
+			`"responseStatus.code"=200 `+
+			`"objectRef.apiGroup"="appstudio.redhat.com" `+
+			`"objectRef.resource"="releases" `+
+			`"objectRef.subresource"="status" `+
+			`"responseObject.metadata.resourceVersion"="*" `+
+			`"responseObject.status.completionTime"="*" `+
+			`| spath responseObject.status.conditions{} `+
+			`| mvexpand responseObject.status.conditions{} `+
+			`| search responseObject.status.conditions{}.type="Released" AND `+
+			`(responseObject.status.conditions{}.reason="Succeeded" OR responseObject.status.conditions{}.reason="Failed") `+
+			`| eval event="Release completed events"`,
```

Comment:
On second thought, I think we should make this dynamic using `responseObject.status.conditions{}.reason`. Let's see what becomes of this thread on the same topic: https://github.com/redhat-appstudio/segment-bridge/pull/34/files#r1312366672.

Also, ` events` should be omitted.